Lib/test/test_shutil.py

```diff
@@ -663,6 +664,29 @@ def test_copytree_named_pipe(self):
             shutil.rmtree(TESTFN, ignore_errors=True)
             shutil.rmtree(TESTFN2, ignore_errors=True)
 
+    @unittest.skipUnless(hasattr(socket, 'AF_UNIX'), 'requires socket.AF_UNIX')
+    def test_copytree_socket(self):
```

Comment:
IMO testing also `copytree()` is not very useful since internally it uses `copyfile` (which you are testing).

Reply:
I was copying what was done for named pipes for consistency. But having a test for `copytree()` makes sense to me because if in the future `copytree()` stopped using `copyfile()` for some reason, we would still want `copytree()` to raise an error?